ckanext/xloader/tests/test_loader.py

```diff
@@ -632,6 +632,18 @@ def test_with_blanks(self, Session):
         )
         assert len(self._get_records(Session, resource_id)) == 3
 
+    def test_with_empty_lines(self, Session):
+        csv_filepath = get_sample_filepath("sample_with_empty_lines.csv")
+        resource = factories.Resource()
+        resource_id = resource['id']
+        loader.load_csv(
+            csv_filepath,
+            resource_id=resource_id,
+            mimetype="text/csv",
+            logger=logger,
+        )
+        assert len(self._get_records(Session, resource_id)) == 6
```

Comment:
can we assert if direct load ran vs backfill tabulator.

Reply:
Not relevant to unit tests. The fallback logic is in jobs.py, unit tests bypass that to directly call load_csv or load_table.